astropy/cosmology/core.py

```diff
@@ -2562,6 +2867,13 @@ def _float_or_none(x, digits=3):
     fmtstr = "{0:.{digits}g}".format(x, digits=digits)
     return fmtstr.format(x)
 
+def vectorize_if_needed(func, x):
+    """ Helper function to vectorize functions on array inputs"""
+    if not isiterable(x):
+        return func(x)
+    else:
+        return np.vectorize(func)(x)
+
```

Comment:
It will be much faster for scalars if we assume a scalar and catch the exception instead of  using `isiterable`:
```
def vectorize_if_needed(func, x):
    """ Helper function to vectorize functions on array inputs"""
    try:
        return func(x)
    except TypeError:
        return np.vectorize(func)(x)
```
This will be a little slower for array inputs. But I think that's ok, because in that case there typically won't be a large number of function calls.